aspnet/security/authorization/secure-data.rst

```diff
@@ -0,0 +1,369 @@
+Create an app with secure user data
+======================================
+
+By `Barry Dorrans`_ and `Rick Anderson`_
+
+.. contents:: Sections:
+  :local:
+  :depth: 1
+
+This tutorial shows how to create a web app with secure user data. Authenticated users can read all the contacts but can only edit their own contacts.  A user in the administrator role can delete any contact.
+
+In the image below, user *rick@example.com* can edit and delete his contacts, and read other contacts.
+
+.. image:: secure-data/_static/conRick.png
+
+In the image below, *test@example.com* is in the ``canDelete`` role, so she can edit her contacts and delete any contact.
+
+.. image:: secure-data/_static/conTest.png
+
+The app was created by scaffolding the following ``Contact`` model:
+
+.. literalinclude::  secure-data/samples/starter/Models/Contact.cs
+  :language: c#
+  :start-after: #region snippet_1
+  :end-before: #endregion
+  :dedent: 4
+  :emphasize-lines: 3
+
+The contact information properties (Address, Name, etc.) are displayed in the images above. ``ContactId`` is the primary key for the table.
+
+A user authorization filter ensures only the logged in user can edit their data. A ``canDelete`` authorization filter allows users in the "canDelete" role to delete any data.
```

Comment:
Filter? How old school. 

An authorization handler ensures that data can only be edited by the data owner. A canDelete handler allows administrators to delete any data.